### ISSUE TITLE
Catmandu installation failed since last update

### ISSUE THREAD

**Issue by Hero-Solutions** [LibreCat/Catmandu#390]
Up until recently, I never had any issues installing Catmandu through cpanminus. However, since a week or so ago (presumably since the last update of Catmandu), any installation attempt fails with the following errors:

```
#   Failed test 'syntax errors throw FixParseError'
#   at t/Catmandu-Fix-Parser.t line 177.
# expecting: Catmandu::FixParseError
# found: Can't locate object method "isa" via package "Parser::MGC::Failure" at /home/vkc/.cpanm/work/1647870888.274878/Catmandu-1.2018/blib/lib/Catmandu/Util.pm line 517.

#   Failed test 'threw Catmandu::FixParseError'
#   at t/Catmandu-Fix-Parser.t line 200.
# expecting: Catmandu::FixParseError
# found: Can't locate object method "isa" via package "Parser::MGC::Failure" at /home/vkc/.cpanm/work/1647870888.274878/Catmandu-1.2018/blib/lib/Catmandu/Util.pm line 517.

#   Failed test 'threw Catmandu::FixParseError'
#   at t/Catmandu-Fix-Parser.t line 208.
# expecting: Catmandu::FixParseError
# found: Can't locate object method "isa" via package "Parser::MGC::Failure" at /home/vkc/.cpanm/work/1647870888.274878/Catmandu-1.2018/blib/lib/Catmandu/Util.pm line 517.
# Looks like you failed 3 tests of 65.
t/Catmandu-Fix-Parser.t ..................
Dubious, test returned 3 (wstat 768, 0x300)
Failed 3/65 subtests
```

This is on a Debian 11 machine hosted on DigitalOcean (thus using the digitalocean repositories for most packages). The installed version of perl is 5.32.1. Perl modules are installed for the local user only (in /home/vkc/), not through root.

Full install log:
```
cpanm (App::cpanminus) 1.7044 on perl 5.032001 built for x86_64-linux-gnu-thread-multi
Work directory is /home/vkc/.cpanm/work/1647871447.277529
You have make /usr/bin/make
You have /usr/bin/wget
You have /usr/bin/tar: tar (GNU tar) 1.34
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.
You have /usr/bin/unzip
!
! Can't write to /usr/local/share/perl/5.32.1 and /usr/local/bin: Installing modules to /home/vkc/perl5
! To turn off this warning, you have to do one of the following:
!   - run me as a root or with --sudo option (to install to /usr/local/share/perl/5.32.1 and /usr/local/bin)
!   - Configure local::lib in your existing shell to set PERL_MM_OPT etc.
!   - Install local::lib by running the following commands
!
!         cpanm --local-lib=~/perl5 local::lib && eval $(perl -I ~/perl5/lib/perl5/ -Mlocal::lib)
!
Searching Catmandu () on cpanmetadb ...
--> Working on Catmandu
Fetching http://www.cpan.org/authors/id/N/NI/NICS/Catmandu-1.2018.tar.gz
-> OK
Unpacking Catmandu-1.2018.tar.gz
Entering Catmandu-1.2018
Checking configure dependencies from META.json
Checking if you have ExtUtils::Install 1.46 ... Yes (2.14)
Checking if you have Module::Build 0.38 ... Yes (0.4231)
Configuring Catmandu-1.2018
Running Build.PL
Created MYMETA.yml and MYMETA.json
Creating new 'Build' script for 'Catmandu' version '1.2018'
-> OK
Checking dependencies from MYMETA.json ...
Checking if you have Any::URI::Escape 0 ... Yes (0.01)
Checking if you have List::MoreUtils 0.401 ... Yes (0.430)
Checking if you have Data::Util 0.66 ... Yes (0.66)
Checking if you have LWP::UserAgent 0 ... Yes (6.57)
Checking if you have YAML::XS 0.41 ... Yes (0.83)
Checking if you have HTTP::Request 0 ... Yes (6.28)
Checking if you have Hash::Merge::Simple 0 ... Yes (0.051)
Checking if you have Sub::Exporter 0.982 ... Yes (0.987)
Checking if you have Role::Tiny::With 0 ... Yes (2.002004)
Checking if you have Module::Info 0 ... Yes (0.37)
Checking if you have IO::Handle::Util 0.01 ... Yes (0.02)
Checking if you have Log::Any::Test 1.03 ... Yes (1.710)
Checking if you have URI::Template 0.22 ... Yes (0.24)
Checking if you have Class::Method::Modifiers 0 ... Yes (2.13)
Checking if you have Moo 1.004006 ... Yes (2.004004)
Checking if you have String::CamelCase 0 ... Yes (0.04)
Checking if you have Log::Log4perl 0 ... Yes (1.54)
Checking if you have Clone 0.31 ... Yes (0.45)
Checking if you have CGI::Expand 2.02 ... Yes (2.05)
Checking if you have MooX::Aliases 0.001006 ... Yes (0.001006)
Checking if you have Text::CSV 1.21 ... Yes (2.01)
Checking if you have Try::Tiny::ByClass 0.01 ... Yes (0.01)
Checking if you have Unicode::Normalize 0 ... Yes (1.27)
Checking if you have Sub::Quote 0 ... Yes (2.006006)
Checking if you have Test::Exception 0.43 ... Yes (0.43)
Checking if you have Role::Tiny 0 ... Yes (2.002004)
Checking if you have Cpanel::JSON::XS 3.0213 ... Yes (4.26)
Checking if you have App::Cmd 0.33 ... Yes (0.334)
Checking if you have MIME::Types 0 ... Yes (2.21)
Checking if you have Path::Tiny 0 ... Yes (0.118)
Checking if you have Test::LWP::UserAgent 0 ... Yes (0.034)
Checking if you have Path::Iterator::Rule 0 ... Yes (1.015)
Checking if you have URI 0 ... Yes (5.08)
Checking if you have Test::More 0.99 ... Yes (1.302188)
Checking if you have Capture::Tiny 0 ... Yes (0.48)
Checking if you have Test::Pod 0 ... Yes (1.52)
Checking if you have Config::Onion 1.004 ... Yes (1.007)
Checking if you have UUID::Tiny 1.04 ... Yes (1.04)
Checking if you have asa 0 ... Yes (1.04)
Checking if you have namespace::clean 0.24 ... Yes (0.27)
Checking if you have Package::Stash 0 ... Yes (0.39)
Checking if you have Module::Build 0.4229 ... Yes (0.4231)
Checking if you have Throwable 0.200004 ... Yes (1.000)
Checking if you have Log::Any::Adapter 0 ... Yes (1.710)
Checking if you have Data::Compare 1.22 ... Yes (1.27)
Checking if you have Parser::MGC 0.21 ... Yes (0.21)
Checking if you have Text::Hogan::Compiler 1.02 ... Yes (2.03)
Checking if you have Log::Any::Adapter::Log4perl 0 ... Yes (0.09)
Checking if you have Time::HiRes 0 ... Yes (1.9764)
Checking if you have Log::Any 0 ... Yes (1.710)
Checking if you have Test::Deep 0.112 ... Yes (1.130)
Building and testing Catmandu-1.2018
Building Catmandu
t/Catmandu-Addable.t ..................... ok
t/Catmandu-ArrayIterator.t ............... ok
t/Catmandu-Bag.t ......................... ok
t/Catmandu-Buffer.t ...................... ok
t/Catmandu-CLI.t ......................... ok
t/Catmandu-CQLSearchable.t ............... ok
t/Catmandu-Cmd-compile.t ................. ok
t/Catmandu-Cmd-config.t .................. ok
t/Catmandu-Cmd-convert.t ................. ok
t/Catmandu-Cmd-copy.t .................... ok
t/Catmandu-Cmd-count.t ................... ok
t/Catmandu-Cmd-delete.t .................. ok
t/Catmandu-Cmd-export.t .................. ok
t/Catmandu-Cmd-help.t .................... ok
t/Catmandu-Cmd-import.t .................. ok
t/Catmandu-Cmd-info.t .................... ok
t/Catmandu-Cmd-run.t ..................... ok
t/Catmandu-Cmd-touch.t ................... ok
t/Catmandu-Cmd.t ......................... ok
t/Catmandu-Counter.t ..................... ok
t/Catmandu-Env.t ......................... ok
t/Catmandu-Error.t ....................... ok
t/Catmandu-Exporter-CSV.t ................ ok
t/Catmandu-Exporter-Count.t .............. ok
t/Catmandu-Exporter-JSON.t ............... ok
t/Catmandu-Exporter-Mock.t ............... ok
t/Catmandu-Exporter-Multi.t .............. ok
t/Catmandu-Exporter-Null.t ............... ok
t/Catmandu-Exporter-TSV.t ................ ok
t/Catmandu-Exporter-Text.t ............... ok
t/Catmandu-Exporter-YAML.t ............... ok
t/Catmandu-Exporter.t .................... ok
t/Catmandu-Fix-Base.t .................... ok
t/Catmandu-Fix-Bind-Group.t .............. ok
t/Catmandu-Fix-Bind-benchmark.t .......... ok
t/Catmandu-Fix-Bind-each.t ............... ok
t/Catmandu-Fix-Bind-hashmap.t ............ ok
t/Catmandu-Fix-Bind-identity.t ........... ok
t/Catmandu-Fix-Bind-importer.t ........... ok
t/Catmandu-Fix-Bind-iterate.t ............ ok
t/Catmandu-Fix-Bind-list.t ............... ok
t/Catmandu-Fix-Bind-maybe.t .............. ok
t/Catmandu-Fix-Bind-timeout.t ............ ok
t/Catmandu-Fix-Bind-visitor.t ............ ok
t/Catmandu-Fix-Bind-with.t ............... ok
t/Catmandu-Fix-Bind.t .................... ok
t/Catmandu-Fix-Condition-all_equal.t ..... ok
t/Catmandu-Fix-Condition-all_match.t ..... ok
t/Catmandu-Fix-Condition-any_equal.t ..... ok
t/Catmandu-Fix-Condition-any_match.t ..... ok
t/Catmandu-Fix-Condition-exists.t ........ ok
t/Catmandu-Fix-Condition-greater_than.t .. ok
t/Catmandu-Fix-Condition-in.t ............ ok
t/Catmandu-Fix-Condition-is_array.t ...... ok
t/Catmandu-Fix-Condition-is_false.t ...... ok
t/Catmandu-Fix-Condition-is_null.t ....... ok
t/Catmandu-Fix-Condition-is_number.t ..... ok
t/Catmandu-Fix-Condition-is_object.t ..... ok
t/Catmandu-Fix-Condition-is_string.t ..... ok
t/Catmandu-Fix-Condition-is_true.t ....... ok
t/Catmandu-Fix-Condition-less_than.t ..... ok
t/Catmandu-Fix-Condition-valid.t ......... ok
t/Catmandu-Fix-Condition.t ............... ok
t/Catmandu-Fix-Inlineable.t .............. ok

#   Failed test 'syntax errors throw FixParseError'
#   at t/Catmandu-Fix-Parser.t line 177.
# expecting: Catmandu::FixParseError
# found: Can't locate object method "isa" via package "Parser::MGC::Failure" at /home/vkc/.cpanm/work/1647871447.277529/Catmandu-1.2018/blib/lib/Catmandu/Util.pm line 517.

#   Failed test 'threw Catmandu::FixParseError'
#   at t/Catmandu-Fix-Parser.t line 200.
# expecting: Catmandu::FixParseError
# found: Can't locate object method "isa" via package "Parser::MGC::Failure" at /home/vkc/.cpanm/work/1647871447.277529/Catmandu-1.2018/blib/lib/Catmandu/Util.pm line 517.

#   Failed test 'threw Catmandu::FixParseError'
#   at t/Catmandu-Fix-Parser.t line 208.
# expecting: Catmandu::FixParseError
# found: Can't locate object method "isa" via package "Parser::MGC::Failure" at /home/vkc/.cpanm/work/1647871447.277529/Catmandu-1.2018/blib/lib/Catmandu/Util.pm line 517.
# Looks like you failed 3 tests of 65.
t/Catmandu-Fix-Parser.t ..................
Dubious, test returned 3 (wstat 768, 0x300)
Failed 3/65 subtests
t/Catmandu-Fix-add_field.t ............... ok
t/Catmandu-Fix-add_to_exporter.t ......... ok
t/Catmandu-Fix-add_to_store.t ............ ok
t/Catmandu-Fix-append.t .................. ok
t/Catmandu-Fix-array.t ................... ok
t/Catmandu-Fix-assoc.t ................... ok
t/Catmandu-Fix-capitalize.t .............. ok
t/Catmandu-Fix-clone.t ................... ok
t/Catmandu-Fix-code.t .................... ok
t/Catmandu-Fix-collapse.t ................ ok
t/Catmandu-Fix-compact.t ................. ok
t/Catmandu-Fix-copy.t .................... ok
t/Catmandu-Fix-copy_field.t .............. ok
t/Catmandu-Fix-count.t ................... ok
t/Catmandu-Fix-cp.t ...................... ok
t/Catmandu-Fix-disassoc.t ................ ok
t/Catmandu-Fix-downcase.t ................ ok
t/Catmandu-Fix-error.t ................... ok
t/Catmandu-Fix-eval.t .................... ok
t/Catmandu-Fix-expand.t .................. ok
t/Catmandu-Fix-expand_date.t ............. ok
t/Catmandu-Fix-export_to_string.t ........ ok
t/Catmandu-Fix-filter.t .................. ok
t/Catmandu-Fix-flatten.t ................. ok
t/Catmandu-Fix-format.t .................. ok
t/Catmandu-Fix-from_json.t ............... ok
t/Catmandu-Fix-hash.t .................... ok
t/Catmandu-Fix-import_from_string.t ...... ok
t/Catmandu-Fix-include.t ................. ok
t/Catmandu-Fix-index.t ................... ok
t/Catmandu-Fix-int.t ..................... ok
t/Catmandu-Fix-join_field.t .............. ok
t/Catmandu-Fix-log.t ..................... ok
t/Catmandu-Fix-lookup.t .................. ok
t/Catmandu-Fix-lookup_in_store.t ......... ok
t/Catmandu-Fix-mapping.t ................. ok
t/Catmandu-Fix-move.t .................... ok
t/Catmandu-Fix-move_field.t .............. ok
t/Catmandu-Fix-mv.t ...................... ok
t/Catmandu-Fix-nothing.t ................. ok
t/Catmandu-Fix-parse_text.t .............. ok
t/Catmandu-Fix-paste.t ................... ok
t/Catmandu-Fix-perlcode.t ................ ok
t/Catmandu-Fix-prepend.t ................. ok
t/Catmandu-Fix-random.t .................. ok
t/Catmandu-Fix-reject.t .................. ok
t/Catmandu-Fix-remove.t .................. ok
t/Catmandu-Fix-remove_field.t ............ ok
t/Catmandu-Fix-rename.t .................. ok
t/Catmandu-Fix-replace_all.t ............. ok
t/Catmandu-Fix-retain.t .................. ok
t/Catmandu-Fix-retain_field.t ............ ok
t/Catmandu-Fix-reverse.t ................. ok
t/Catmandu-Fix-rm.t ...................... ok
t/Catmandu-Fix-search_in_store.t ......... ok
t/Catmandu-Fix-set_array.t ............... ok
t/Catmandu-Fix-set_field.t ............... ok
t/Catmandu-Fix-set_hash.t ................ ok
t/Catmandu-Fix-sleep.t ................... ok
t/Catmandu-Fix-sort_field.t .............. ok
t/Catmandu-Fix-split_field.t ............. ok
t/Catmandu-Fix-string.t .................. ok
t/Catmandu-Fix-substring.t ............... ok
t/Catmandu-Fix-sum.t ..................... ok
t/Catmandu-Fix-to_json.t ................. ok
t/Catmandu-Fix-trim.t .................... ok
t/Catmandu-Fix-uniq.t .................... ok
t/Catmandu-Fix-upcase.t .................. ok
t/Catmandu-Fix-uri_decode.t .............. ok
t/Catmandu-Fix-uri_encode.t .............. ok
t/Catmandu-Fix-uri_encode_decode.t ....... ok
t/Catmandu-Fix-vacuum.t .................. ok
t/Catmandu-Fix-validate.t ................ ok
t/Catmandu-Fix.t ......................... ok
t/Catmandu-Fixable.t ..................... ok
t/Catmandu-Flushable.t ................... ok
t/Catmandu-Hits.t ........................ ok
t/Catmandu-IdGenerator-Mock.t ............ ok
t/Catmandu-IdGenerator-UUID.t ............ ok
t/Catmandu-IdGenerator.t ................. ok
t/Catmandu-Importer-CSV.t ................ ok
t/Catmandu-Importer-JSON.t ............... ok
t/Catmandu-Importer-Mock.t ............... ok
t/Catmandu-Importer-Modules.t ............ ok
t/Catmandu-Importer-Multi.t .............. ok
t/Catmandu-Importer-TSV.t ................ ok
t/Catmandu-Importer-Text.t ............... ok
t/Catmandu-Importer-YAML.t ............... ok
t/Catmandu-Importer.t .................... ok
t/Catmandu-Interactive.t ................. ok
t/Catmandu-Iterable.t .................... ok
t/Catmandu-IterableOnce.t ................ ok
t/Catmandu-Iterator.t .................... ok
t/Catmandu-MultiIterator.t ............... ok
t/Catmandu-Paged.t ....................... ok
t/Catmandu-Path-simple.t ................. ok
t/Catmandu-Pluggable.t ................... ok
t/Catmandu-Plugin-Datestamps.t ........... ok
t/Catmandu-Plugin-Readonly.t ............. ok
t/Catmandu-Plugin-Versioning.t ........... ok
t/Catmandu-Sane.t ........................ ok
t/Catmandu-Searchable.t .................. ok
t/Catmandu-Serializer.t .................. ok
t/Catmandu-Store-Hash.t .................. ok
t/Catmandu-Store-Multi-Bag.t ............. ok
t/Catmandu-Store-Multi.t ................. ok
t/Catmandu-Store.t ....................... ok
t/Catmandu-Util.t ........................ ok
t/Catmandu-Validator-Mock.t .............. ok
t/Catmandu-Validator-Simple.t ............ ok
t/Catmandu-Validator.t ................... ok
t/Catmandu.t ............................. ok
t/author-pod-syntax.t .................... skipped: these tests are for testing by the author
t/release-perltidy.t ..................... skipped: these tests are for release candidate testing

Test Summary Report
-------------------
t/Catmandu-Fix-Parser.t                (Wstat: 768 Tests: 65 Failed: 3)
  Failed tests:  41, 48, 51
  Non-zero exit status: 3
Files=179, Tests=2125, 62 wallclock secs ( 0.99 usr  0.53 sys + 45.26 cusr  5.59 csys = 52.37 CPU)
Result: FAIL
Failed 1/179 test programs. 3/2125 subtests failed.
-> FAIL Installing Catmandu failed. See /home/vkc/.cpanm/work/1647871447.277529/build.log for details. Retry with --force to force install it.
```

**Comment by vpeil** [comment]
same problem as  in #389

**Comment by nics** [comment]
hi @Hero-Solutions, it's some weird isa issue, i released 1.2019 which will hopefully fix this

**Comment by Hero-Solutions** [comment]
Installation works again, thank you very much for the quick fix and fast reply!
